tests/fixtures/expect.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fixtures
+
+import (
+	"context"
+	"testing"
+
+	"github.com/numaproj-labs/numaplane/pkg/apis/numaplane/v1alpha1"
+	planepkg "github.com/numaproj-labs/numaplane/pkg/client/clientset/versioned/typed/numaplane/v1alpha1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Expect struct {
+	t             *testing.T
+	restConfig    *rest.Config
+	kubeClient    kubernetes.Interface
+	gitSync       *v1alpha1.GitSync
+	gitSyncClient planepkg.GitSyncInterface
+}
+
+// check that resources are created
+func (e *Expect) ResourcesExist(resourceType string, resources []string) *Expect {
+
+	e.t.Helper()
+	ctx := context.Background()
+	for _, r := range resources {
+		result := e.kubeClient.CoreV1().RESTClient().Get().Timeout(defaultTimeout).Namespace(TargetNamespace).Resource(resourceType).Name(r).Do(ctx)
+		if result.Error() != nil {
+			e.t.Fatalf("Resource %s does not exist", r)
+		}
+	}
+
+	return e
+}
+
+// check that resources are deleted
+func (e *Expect) ResourcesDontExist(resourceType string, resources []string) *Expect {
+
+	e.t.Helper()
+	ctx := context.Background()
+	for _, r := range resources {
+		result := e.kubeClient.CoreV1().RESTClient().Get().Timeout(defaultTimeout).Namespace(TargetNamespace).Resource(resourceType).Name(r).Do(ctx)
```

Comment:
Is this line right? It looks like it might be the same as the function above? (I'm not sure what `Do()` is)

Reply:
Do outputs the result that we check for the Get request. It is the same logic but we're looking for an error to tell us the resources don't exist. This might not be the best approach. Alternatively, we could list Pods and check that those don't exist.

Reply:
I think it won't work with the timeout you're passing. It will check if everything exists, which it initially may if everything's just been deleted.